src/parsy/__init__.py

```diff
@@ -516,6 +513,17 @@ def fail(expected: str) -> Parser:
     return Parser(lambda _, index: Result.failure(index, expected))
 
 
+def make_index_update(consumed: str) -> Callable[[Position], Position]:
```

Comment:
This function is curried to avoid recomputing the `count` and `rfind` methods every single time someone uses the `string` parser